presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriter.java

```diff
@@ -245,13 +246,14 @@ public Optional<Page> commit()
         try {
             orcWriter.close();
         }
-        catch (IOException | UncheckedIOException e) {
+        catch (IOException | UncheckedIOException | PrestoException e) {
             try {
                 rollbackAction.call();
             }
             catch (Exception ignored) {
                 // ignore
             }
+            propagateIfPossible(e.getCause(), PrestoException.class);
```

Comment:
does this work?
propagateIfPossible will match the given exception or a RuntimeException / Error as per javadocs.
1. does `e.getCause` match `PrestoException`?
2. Also what if the actual exception is an UncheckedIOException , since it is a RuntimeException looks like it will match propagateIfPossible's rules and we will not throw a PrestoException back.

Also propagateIfPossible is a [deprecated](https://guava.dev/releases/23.0/api/docs/com/google/common/base/Throwables.html#propagateIfPossible-java.lang.Throwable-) method and scheduled for removal. (I know we use it at other places in presto)

Reply:
1. good catch. it should just be `e` and not `e.getCause()`
2. `[propagateIfPossible](https://guava.dev/releases/23.0/api/docs/com/google/common/base/Throwables.html#propagateIfPossible-java.lang.Throwable-)([Throwable](http://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html?is-external=true) throwable)` is deprecated but `[propagateIfPossible](https://guava.dev/releases/23.0/api/docs/com/google/common/base/Throwables.html#propagateIfPossible-java.lang.Throwable-java.lang.Class-)([Throwable](http://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html?is-external=true) throwable, [Class](http://docs.oracle.com/javase/8/docs/api/java/lang/Class.html?is-external=true)<X> declaredType)` is NOT deprecated.

Reply:
I see the description of propagateIfPossible . It says "Propagates throwable exactly as-is, if and only if it is an instance of RuntimeException, Error, or declaredType."

May be i will use [throwIfInstanceOf](https://guava.dev/releases/23.0/api/docs/src-html/com/google/common/base/Throwables.html#line.72)